enterprise/ha/src/main/java/org/neo4j/kernel/ha/lock/forseti/ForsetiClient.java

```diff
@@ -656,6 +665,17 @@ public int id()
         return myId;
     }
 
+    private void applyWaitStrategy( Locks.ResourceType resourceType, int tries )
```

Comment:
This could lead to `TransactionTerminatedException` being wrapped in `RuntimeException`. Fixed in https://github.com/neo4j/neo4j/pull/7168/commits/058fd6cdaf28b65496812256b9b297c0f5406d1c